### DISCUSSION TITLE
How to handle an image with different kinds of texts (for example, there are printed texts and written words?

### DISCUSSION THREAD

**Discussion by felix115** [PaddlePaddle/PaddleOCR#6019] [Q&A] [answered]
How to handle an image with different kinds of texts (for example, there are printed texts and written words) ?

**Comment by Evezerest** [comment, marked as answer]
This is very common. Take an exam paper as an example. It has both printed texts and written words. Under this circumstance, you can adopt the solution of "1 detection model+ 1 N-class model+ N recognition models".

All types of texts share one detection model, and the N-class model which is a classifier in the extra training classifies the texts. If there are printed and written texts, the model is a two-class model; if there are N types of texts, the model is an N-class model. When it comes to recognition, every text type trains a recognition model. If there are printed and written texts in an image, two recognition models are needed, one for printed texts and the other for written texts. Then, when a text box is classified as a written text, then it will be recognized by the specialized model. The same goes for other cases.
